registration/views/onsite_admin.py

```diff
@@ -1065,25 +1065,29 @@ def create_discount(request):
             amount_off = Decimal(amount[1:])
         elif amount.startswith("%"):
             percent_off = int(amount[1:])
+        elif amount.endswith("%"):
+            percent_off = int(amount[:-1])
+        else:
+            return JsonResponse({"success": False, "reason": "Unknown discount type"}, status=400)
     except ValueError as e:
         return JsonResponse({"success": False, "reason": str(e)}, status=400)
 
     cart = request.session.get("cart", None)
     if cart is None:
         request.session["cart"] = []
         return JsonResponse(
-            {"success": False, "message": "Cart not initialized"}, status=400
+            {"success": False, "reason": "Cart not initialized"}, status=400
         )
 
     discount = Discount(
-        codeName=get_b32_uuid(),
```

Comment:
[get_token](https://github.com/furthemore/APIS/blob/31b2a6b0a0589047c4aaf3d4afb8fb1797e39633/registration/models.py#L372) is not a terribly robust mechanism for getting a unique code.  Anywhere else it's used usually features this pattern:

```python
while (token not in db):
    token = get_token()
```

A UUID should be globally unique, but if you want to `get_token`, we should include a check to handle retrying if there's a collision.

Reply:
Ah yeah, that makes sense. I've swapped back to `get_b32_uuid` for generating the code.